src/librustdoc/html/static/js/storage.js

```diff
@@ -1,5 +1,8 @@
 // From rust:
 /* global resourcesSuffix */
+var resourcesSuffix = /* AUTOREPLACE: */"RESOURCE_SUFFIX";
```

Comment:
I'm a bit worried about this: if we generate docs with two different rustdoc version and/or different suffix, it means the older version will use newer files, which might break the display.

Reply:
storage is stored as a versioned file (e.g. storage1.58.0.js), so it would load the other versioned files that match it.

Reply:
That is, if you use suffix, otherwise you'll simply overwrite.

Reply:
I don't think it's possible for rustdoc alone to prevent conflicting versions with our current setup. If you think that should be a goal, maybe open an issue or MCP but I don't think it makes sense to hold specific parts of rustdoc to a higher standard than others.

Reply:
The more I think about it, the more I think it would be cleaner to put each of these three variables in a `data-` attribute in the HTML, like I proposed in https://github.com/rust-lang/rust/pull/91062#discussion_r753640831. The big advantage there is that when you're editing JS you can copy it directly into place without waiting to rebuild rustdoc. I've run into this a number of times just recently.

Reply:
> I don't think it's possible for rustdoc alone to prevent conflicting versions with our current setup. If you think that should be a goal, maybe open an issue or MCP but I don't think it makes sense to hold specific parts of rustdoc to a higher standard than others.

It is working currently because the links are set directly in the HTML, not in the JS which is regenerated. This is my biggest issue with this change, moving "hard-coded" HTML content into JS.

@jsha Yes, I think that could work. The added size would be negligible as well.